cpp/src/hash/hashing.cu

```diff
@@ -78,15 +78,15 @@ class bitwise_partitioner
 };
 
 template <bool has_nulls>
-std::vector<std::unique_ptr<experimental::table>>
+std::pair<std::unique_ptr<experimental::table>, std::vector<size_type>>
 hash_partition_table(table_view const& input,
                      table_view const &table_to_hash,
                      const size_type num_partitions,
                      rmm::mr::device_memory_resource* mr,
                      cudaStream_t stream)
 {
   auto const num_rows = table_to_hash.num_rows();
```

Comment:
This is the only line where table_to_hash is used in this function.  We are hashing the original input table instead of the table with the columns we care about in it.

Reply:
```diff
diff --git a/cpp/src/hash/hashing.cu b/cpp/src/hash/hashing.cu
index 640ffa966..1c76dac95 100644
--- a/cpp/src/hash/hashing.cu
+++ b/cpp/src/hash/hashing.cu
@@ -89,8 +89,8 @@ hash_partition_table(table_view const& input,
   auto row_partitions = rmm::device_vector<size_type>(num_rows);
 
   // Make an iterator to compute the hash over each row
-  auto const device_input = table_device_view::create(input, stream);
-  auto const hasher = experimental::row_hasher<MurmurHash3_32, has_nulls>(*device_input);
+  auto const device_to_hash_input = table_device_view::create(table_to_hash, stream);
+  auto const hasher = experimental::row_hasher<MurmurHash3_32, has_nulls>(*device_to_hash_input);
   auto const hash_iterator = thrust::make_transform_iterator(
     thrust::make_counting_iterator(0), hasher);
```

fixed the problem for me.

Reply:
Good catch! I'll fix this and add a unit test to make sure that non-hashed columns don't affect the partitioning

Reply:
Fixed in https://github.com/rapidsai/cudf/pull/3701/